lib/ansible/modules/cloud/hcloud/hcloud_server.py

```diff
@@ -0,0 +1,408 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, Hetzner Cloud GmbH <info@hetzner-cloud.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = """
+---
+module: hcloud_server
+
+short_description: Create and manage cloud servers on the Hetzner Cloud
+
+version_added: "2.8"
+
+description:
+    - "Create, update and manage cloud servers on the Hetzner Cloud."
+
+author:
+    - Lukas Kaemmerling (@lkaemmerling)
+
+options:
+    id:
+        description:
+            - The Server ID
+            - Only required if no server name is given
+        type: int
+    name:
+        description:
+            - The Server Name
+            - Only required if no server id is given or a server does not exists
+        type: str
+    server_type:
+        description:
+            - The Server Type
+            - Required if server does not exists
+        type: str
+    ssh_keys:
+        description:
+            - List of SSH Keys Names
+        type: list
+        aliases: [ ssh_key ]
+    volumes:
+        description:
+            - List of Volumes IDs that should be attached to the server on server creation
+        type: list
+    image:
+        description:
+            - Image the server should be created from
+            - Required if server does not exists
+        type: str
+    location:
+        description:
+            - Location of Server
+            - Required if no I(datacenter) is given and server does not exists
+        type: str
+    datacenter:
+        description:
+            - Datacenter of Server
+            - Required of no I(location) is given and server does not exists
+        type: str
+    backups:
+        description:
+            - Enable or disable Backups for the given Server
+        type: bool
+        default: False
+    upgrade_disk:
+        description:
+            - Resize the disk size, when resizing a server.
+            - If you want to downgrade the server later, this value should be False.
+        type: bool
+        default: no
+    force_upgrade:
+        description:
+            - Force the upgrade of the server. Power off the server if it is running on upgrade
+        type: bool
+        default: no
+    user_data:
+        description:
+            - User Data to be passed to the server on creation.
+            - Only used if server does not exists
+        type: str
+    state:
+        description:
+            - State of the server.
+        default: present
+        choices: [ present, absent, restarted, started, stopped ]
```

Comment:
We tend to sort items in lists when the order is of no importance.
This helps people to compare lists, or find an entry more quickly.
The documentation sorts them as well during generation, so you cannot influence the order seen by the user.
```suggestion
        choices: [ absent, present, restarted, started, stopped ]
```